docs/contributing/code/setting-up-the-dev-environment.md

```diff
@@ -0,0 +1,179 @@
+# Setting up the Development Environment
+
+This document guides you through setting up your development environment to start contributing to our codebase.
+
+:::important Prerequisites
+
+Please complete our [Introduction](introduction/getting-started.mdx) guides before proceeding. This ensures you have the necessary tools and dependencies installed.
+
+:::
+
+## Setting Up The Monorepo Management
+
+Our repository is a monorepo, which means it contains multiple Detox-related projects and packages. [Read more about our repository structure](../code/overview.md#repository-structure).
+
+To set up the monorepo locally, follow these steps:
+
+Install the monorepo management tool, `lerna`:
+
+```bash npm2yarn
+npm install lerna@3.x.x --global
+```
+
+Clone the repository and navigate to the project directory:
+
+```bash
+git clone git@github.com:wix/Detox.git
+cd detox
+git submodule update --init --recursive
+```
+
+From the project's root directory, install and link the internal projects:
+
+```bash
+lerna bootstrap
+```
+
+## Installing Common Dependencies
+
+### React-Native CLI
+
+[react-native-cli] is a command line interface for React Native.
+
+```bash npm2yarn
+npm install react-native-cli --global
+```
+
+### Watchman
+
+[Watchman] is a tool by Facebook for watching changes in the filesystem.
+
+```bash
+brew install watchman
+```
+
+### xcpretty
+
+[xcpretty] is a fast and flexible formatter for xcodebuild.
+
+```bash
+gem install xcpretty
+```
+
+## Building and Testing
```

Comment:
This has to be split away onto a dedicate guide, as it's not about setting up the environment.